core/domain/state_domain.py

```diff
@@ -608,6 +608,9 @@ def get_all_html_content_strings(self):
             list(str): The list of all html content strings in the interaction.
         """
         html_list = []
+        if self.id is None:
+            return html_list
```

Comment:
Can you please check whether we flush hints after deleting interactions? (I think we keep it just in case if the users readds the same interaction.) Can you please cehk and update it accordingly?

Reply:
Yeah, I checked it and the state still keeps the hints from previous interaction even if we delete the interaction. Thanks for raising this.
Changed the code.